### DISCUSSION TITLE
Use automerge feature of GitLab

### DISCUSSION THREAD

**Discussion by mcfedr** [renovatebot/renovate#15616] [Request Help] [answered]
### What would you like Renovate to be able to do?

When making MRs for changes that are configured for auto merge by renovate it would be great if it set the 'merge when pipeline succeeds' flag.

This would mean the changes are merged quicker without another run of renovate required to merge them.

### If you have any ideas on how this should be implemented, please tell us here.

I have not looked at what would be required to implement it. 

I know that GitLab API allows setting this flag once a MR has been created.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment, marked as answer]
Do you mean https://docs.renovatebot.com/configuration-options/#platformautomerge ?

**Comment by mcfedr** [comment]
Ha, I guess you are right, who knew...

would it be worth enabling this by default in say the gitlab renovate runner?

**Reply by rarkins** [reply]
Always check the docs first, e.g.

![image](https://user-images.githubusercontent.com/6311784/168803239-e15c5404-e732-4b23-95b3-3b87000e5b1f.png)

And don't forget the option above it, if you need it!

I think it's best to leave opt-in for now, but we may switch it in a major release later. We don't make a feature default enabled if it would be unwelcome or surprisingly to more than a small percentage of users.
